spec/codeowners/checker_spec.rb

```diff
@@ -297,4 +341,53 @@ def setup_git_for_project
       expect(subject.main_group).to be_a(Codeowners::Checker::Group)
     end
   end
+
+  describe '#valid_owner?' do
+    subject { described_class.new folder_name, from, to }
+
+    before do
+      subject.owners_list
+    end
+
+    context 'when user is in teams.txt' do
+      it 'returns true' do
+        expect(subject.valid_owner?('@owner1')).to eq(true)
```

Comment:
We also have two things to improve here, one is use predicated methods:

```ruby
        expect(subject).to be_valid_owner('@owner1')
```

and you can also switch `expect(subject)` to `is_expected`:

```ruby
        is_expected.to be_valid_owner('@owner1')
```
I'd simplify the entire block to one single spec:

```ruby
    context 'when load teams.txt' do
      it 'validates owner' do
        is_expected.to be_valid_owner('@owner1')
        is_expected.not_to be_valid_owner('@unkown')
      end
    end

    context 'when skip validations' do
      before do
        subject.validate_owners = false
      end
      it { is_expected.to be_valid_owner('@unkown') }
    end
```